### PR TITLE
INTEG-225: Add DynamoDB Client

### PR DESCRIPTION
**Pull request by ghost**
## Purpose

Follow up PR to #2812

## Approach

-  Adds DynamoDB client using the `@aws-sdk/client-dynamodb` and `@aws-sdk/lib-dynamodb` 
-  Adds encrypt/decrypt methods for serializing and storing service keys

## Dependencies and/or References

closes: https://contentful.atlassian.net/browse/INTEG-225, https://contentful.atlassian.net/browse/INTEG-226

## Deployment

Part of an incremental adoption plan over successive PRS

### CODE REVIEW

File: apps/google-analytics-4/lambda/src/services/awsApiService.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    try {
+      const command = new PutCommand({
+        TableName: input.tableName,
+        Item: {
+          sharedCredentialsId: input.sharedCredentialsId,
+          value: hash, // Add encrypt/decrypt methods
+        },
+      });
+
+      const data = await this.dynamoDbClient.send(command);
```
Comment (jsdalton) [inline apps/google-analytics-4/lambda/src/services/awsApiService.ts:52]: I don't really see a huge problem with the way you've done this, but any reason not to use the `put()` method per the documentation as opposed to the low level `send()`? https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html#put-property

File: apps/google-analytics-4/lambda/src/services/dynamoDbService.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    client = new DynamoDBClient({
+      region: config.awsRegion,
+      endpoint: config.dynamoDbEndpoint,
+    })
+  ) {
+    this.dynamoDBDocumentClient = DynamoDBDocumentClient.from(client);
+    this.tableName = config.dynamoDbTableName;
+  }
+
+  async getSharedCredentials(input: SharedCredentialsInput) {
```
Comment (jsdalton) [inline apps/google-analytics-4/lambda/src/services/dynamoDbService.ts:25]: Take this as a follow up PR but I'm not seeing the value of sharing `input` between these two methods.

This is a pretty common pattern for getters and setters where you have:

get(key)
set(key, value)

And i think ideally what you're doing is much the same:

getSharedCredentials(sharedCredentialsId)
saveSharedCredentials(sharedCredentialsId, sharedCredentials)

**Review by jsdalton — APPROVED** [review]
Looking good, it's progress. I called out a small thing for a subsequent PR but nothing blocking.

File: apps/google-analytics-4/lambda/src/services/dynamoDbService.ts (hunk 2 above)
Comment (gordo-contetnful) [reply apps/google-analytics-4/lambda/src/services/dynamoDbService.ts:25]: You are spot on, I will add this cleanup in the client side PR coming tomorrow 👍

File: apps/google-analytics-4/lambda/src/controllers/apiController.ts [hunk 3]
```diff
@@ -12,6 +13,25 @@ const ApiController = {
...
+
+    // if (!sharedCredentials) {
+    //   const data = await dynamoDB.saveSharedCredentials({
+    //     sharedCredentialsId,
+    //     serviceKey: req.serviceAccountKey!,
+    //   });
+    //   console.log({ data });
+    // } else {
+    //   console.log({ sharedCredentials });
+    // }
```
Comment (ghepting) [inline apps/google-analytics-4/lambda/src/controllers/apiController.ts:33]: remove all of this unused psuedo code. we have the final product in another branch

**Pull request merged**